src/widget/textbox.rs

```diff
@@ -431,12 +431,13 @@ impl Widget<String> for TextBoxRaw {
                         self.reset_cursor_blink(ctx);
                     }
                     // Actual typing
-                    k_e if k_e.key_code.is_printable() => {
```

Comment:
The problem with removing this is that control characters get inserted into the string. I agree with you that we shouldn't be looking at the keycode (that's not really a source of truth), but should have an API indicating whether the keyboard event has printable text.

Reply:
In what situations do control characters come through this path? I haven’t tried much, but nothing immediately sprang to mind.

Reply:
It happens when you press, say, control-A, but the command isn't interpreted.